src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User registerUser(String email, String password) throws RegistrationException {
+        Optional<User> user = userService.findByEmail(email);
+        if (user.isEmpty()) {
+            User newUser = new User();
+            newUser.setEmail(email);
+            newUser.setSalt(HashUtil.getSalt());
+            newUser.setPassword(HashUtil.hashPassword(password, newUser.getSalt()));
```

Comment:
we set salt and hash password in `add()` method, lets remove it from registerUser method